eclair-core/src/main/scala/fr/acinq/eclair/payment/send/PaymentLifecycle.scala

```diff
@@ -198,19 +203,30 @@ class PaymentLifecycle(nodeParams: NodeParams, cfg: SendPaymentConfig, router: A
       case Success(e@Sphinx.DecryptedFailurePacket(nodeId, failureMessage: Update)) =>
         log.info(s"received 'Update' type error message from nodeId=$nodeId, retrying payment (failure=$failureMessage)")
         val failure = RemoteFailure(cfg.fullRoute(route), e)
-        val ignore1 = if (Announcements.checkSig(failureMessage.update, nodeId)) {
+        if (Announcements.checkSig(failureMessage.update, nodeId)) {
           val assistedRoutes1 = handleUpdate(nodeId, failureMessage, d)
           val ignore1 = PaymentFailure.updateIgnored(failure, ignore)
           // let's try again, router will have updated its state
-          router ! RouteRequest(nodeParams.nodeId, c.targetNodeId, c.finalPayload.amount, c.getMaxFee(nodeParams), assistedRoutes1, ignore1, c.routeParams, paymentContext = Some(cfg.paymentContext))
-          ignore1
+          c match {
+            case (_: SendPaymentToRoute) =>
+              log.error("unexpected retry during SendPaymentToRoute")
```

Comment:
This log line doesn't make sense, it's completely ok to receive a remote error here, we're not retrying.
Maybe just replace it with:
```suggestion
              log.info("not retrying SendPaymentToRoute")
```

Reply:
It is fine to receive a remote error but this is still a form of retry. There is already this case that prevents retries and takes care of such remote errors:

```scala
case res if failures.size + 1 >= c.maxAttempts =>
```

For `SendPaymentToRoute`, `maxAttempts` is set to 1 so we should never reach this retry.

Reply:
Good catch, you're right!